lib/fog/ovirt/requests/compute/v4/update_vm.rb

```diff
@@ -0,0 +1,45 @@
+module Fog
+  module Compute
+    class Ovirt
+      class V4
+        class Real
+          # rubocop:disable Metrics/AbcSize
+          def update_vm(attrs)
+            attrs = attrs.dup
+            vm_service = client.system_service.vms_service.vm_service(attrs[:id])
+
+            if attrs[:cores].present?
+              cpu_topology = OvirtSDK4::CpuTopology.new(:cores => attrs[:cores], :sockets => "1")
+              attrs[:cpu] = OvirtSDK4::Cpu.new(:topology => cpu_topology)
+            end
+            attrs[:cluster] = attrs[:cluster].present? ? client.system_service.clusters_service.cluster_service(attrs[:cluster]).get : nil
+            attrs[:template] = if attrs[:template].present?
+                                 client.system_service.templates_service.template_service(attrs[:template]).get
+                               else
+                                 client.system_service.get.special_objects.blank_template
+                               end
+
+            attrs[:large_icon] = OvirtSDK4::Icon.new(:id => attrs[:large_icon])
+            attrs[:small_icon] = OvirtSDK4::Icon.new(:id => attrs[:small_icon])
+            attrs[:cpu_profile] = client.system_service.cpu_profiles_service.profile_service(attrs[:cpu_profile]).get
+            attrs[:host] = client.system_service.hosts_service.host_service(attrs[:host]).get
+            attrs[:original_template] = if attrs[:original_template].present?
+                                          client.system_service.templates_service.template_service(attrs[:original_template]).get
+                                        else
+                                          client.system_service.get.special_objects.blank_template
+                                        end
+            vm_service.update(attrs)
```

Comment:
hen testing, I got:

```
2018-07-03T12:19:14 [W|app|d638a] Failed to update a compute abcde ovirt (oVirt) instance inecas-test-19-ovirt.inecas.ovirt.test: undefined method `href' for "#<OvirtSDK4::Bios:0x0055cd67e67320>":String
 | NoMethodError: undefined method `href' for "#<OvirtSDK4::Bios:0x0055cd67e67320>":String
 | /home/inecas/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ovirt-engine-sdk-4.2.4/lib/ovirtsdk4/writers.rb:496:in `write_one'
 | /home/inecas/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ovirt-engine-sdk-4.2.4/lib/ovirtsdk4/writers.rb:7023:in `write_one'
 | /home/inecas/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ovirt-engine-sdk-4.2.4/lib/ovirtsdk4/writer.rb:213:in `call'
 | /home/inecas/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ovirt-engine-sdk-4.2.4/lib/ovirtsdk4/writer.rb:213:in `write'
 | /home/inecas/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ovirt-engine-sdk-4.2.4/lib/ovirtsdk4/service.rb:289:in `internal_update'
 | /home/inecas/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/ovirt-engine-sdk-4.2.4/lib/ovirtsdk4/services.rb:29194:in `update'
 | /home/inecas/Projects/ws/foreman-rex/fog-ovirt/lib/fog/ovirt/requests/compute/v4/update_vm.rb:31:in `update_vm'
 | /home/inecas/Projects/ws/foreman-rex/fog-ovirt/lib/fog/ovirt/models/compute/server.rb:177:in `save'
```

Probably missed conversion after extracting from the cloudinit PR.